SCSH_3C_iOS/SCSH_3C_iOS/Modules/Explore/View/ExploreViewController.swift

```diff
@@ -0,0 +1,139 @@
+        switch section {
+        case SectionName.Board.rawValue, SectionName.Recent.rawValue, SectionName.Brand.rawValue, SectionName.Popular.rawValue:
+            return 1
+        default:
+            return 15
+        }
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        .init(top: 10, left: 0, bottom: 0, right: 0)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        switch indexPath.section {
+        case SectionName.Board.rawValue:
+            return CGSize(width: collectionView.bounds.width, height: 220)
+            
+        case SectionName.Recent.rawValue:
+            return CGSize(width: collectionView.bounds.width, height: 200)
+
+        case SectionName.Brand.rawValue:
+            return CGSize(width: collectionView.bounds.width, height: 300)
+
+        case SectionName.Popular.rawValue:
+            return CGSize(width: collectionView.bounds.width, height: 150)
+
+        default:
+            return CGSize(width: collectionView.bounds.width / 2 - 10, height: 200)
+        }
+    }
+}
+
+extension ExploreViewController: UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        
+        switch indexPath.section {
+        case SectionName.Board.rawValue:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "BoardSectionCollectionViewCell", for: indexPath) as! BoardSectionCollectionViewCell
+            return cell
+            
+        case SectionName.Recent.rawValue:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "RecentSectionCollectionViewCell", for: indexPath) as! RecentSectionCollectionViewCell
+            return cell
+            
+        case SectionName.Brand.rawValue:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "BrandSectionCollectionViewCell", for: indexPath) as! BrandSectionCollectionViewCell
+            return cell
+            
+        case SectionName.Popular.rawValue:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "PopularSectionCollectionViewCell", for: indexPath) as! PopularSectionCollectionViewCell
+            return cell
+            
+        case SectionName.Explore.rawValue:
+            let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "ExploreSectionCollectionViewCell", for: indexPath) as! ExploreSectionCollectionViewCell
+            cell.contentView.backgroundColor = .blue
+            return cell
+            
+        default:
+            return UICollectionViewCell()
+        }
```

Comment:
Do you think the code would be cleaner if eliminate 'return cell' from each case and have a single return statement after the switch?

Or it is impossible to declare `cell` using a generic(interface) type?

Reply:
> Do you think the code would be cleaner if eliminate 'return cell' from each case and have a single return statement 
> after the switch?

I can't get the casting type at the compile-time. ( Static / Dynamic type issue )

**For example**
```
switch indexPath.section {
        let cell: UICollectionViewCell
            
        case SectionName.Board.rawValue:
            cell = collectionView.dequeueReusableCell(withReuseIdentifier: "BoardSectionCollectionViewCell", for: indexPath) as! BoardSectionCollectionViewCell
           
           /// CATN'T get type `BoardSectionCollectionViewCell` at the compile-time.
           
        default:
            cell = UICollectionViewCell()
        }
           return cell
}
```

Reply:
I got it.

Reply:
> Or it is impossible to declare cell using a generic(interface) type?

I'm not quite sure, but I prefer to implement UI in a straightforward way at the beginning.